modules/root/root_api.rb

```diff
@@ -25,4 +27,19 @@ class Proxy::RootApi < Sinatra::Base
       log_halt 400, e
     end
   end
+
+  get "/events" do
+    begin
+      content_type :json
+
+      events = []
+      from_serial = params[:from_serial].to_i rescue 0
```

Comment:
nitpick: probably rescue is not required as nil.to_i and ''.to_i in both cases will be 0.